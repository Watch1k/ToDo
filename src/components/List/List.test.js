import React from 'react'
import { render } from '@testing-library/react'
import ToDoList from './'

const todos = [
  {
    text: 'todo1',
    completed: false,
    id: new Date().getTime(),
  },
  {
    text: 'todo2',
    completed: true,
    id: new Date().getTime() + 1,
  },
]

describe('List component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = render(<ToDoList todos={[...todos]} />)
  })

  it('Renders', () => {
    expect(wrapper.getByTestId('list')).toBeInTheDocument()
  })

  it('Matches snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('Renders all items', () => {
    expect(wrapper.getAllByTestId('list-item')).toHaveLength(todos.length)
  })
})
