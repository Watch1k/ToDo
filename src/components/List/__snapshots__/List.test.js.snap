// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-list ant-list-split"
    data-testid="list"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <li
          class="ant-list-item"
          data-testid="list-item"
        >
          <div
            class="ant-list-item-meta"
          >
            <div
              class="ant-list-item-meta-content"
            >
              <h4
                class="ant-list-item-meta-title"
              >
                <p
                  data-testid="item-p"
                  style="margin: 0px;"
                >
                  New task #1
                </p>
              </h4>
            </div>
          </div>
          <ul
            class="ant-list-item-action"
          >
            <li>
              <button
                class="ant-btn ant-btn-icon-only"
                data-testid="item-edit"
                style="color: rgb(255, 255, 255); background: rgb(124, 179, 5);"
                type="button"
              >
                <i
                  aria-label="icon: edit"
                  class="anticon anticon-edit"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="edit"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 0 0 0-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 0 0 9.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                    />
                  </svg>
                </i>
              </button>
              <em
                class="ant-list-item-action-split"
              />
            </li>
            <li>
              <button
                class="ant-btn ant-btn-icon-only"
                style="color: rgb(255, 255, 255); background: rgb(255, 120, 117);"
                type="button"
              >
                <i
                  aria-label="icon: delete"
                  class="anticon anticon-delete"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="delete"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                    />
                  </svg>
                </i>
              </button>
              <em
                class="ant-list-item-action-split"
              />
            </li>
            <li>
              <button
                class="ant-btn ant-btn-link"
                style="font-size: 20px;"
                type="button"
              >
                <i
                  aria-label="icon: check-circle"
                  class="anticon anticon-check-circle"
                  style="color: rgb(217, 217, 217);"
                  tabindex="-1"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="check-circle"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M699 353h-46.9c-10.2 0-19.9 4.9-25.9 13.3L469 584.3l-71.2-98.8c-6-8.3-15.6-13.3-25.9-13.3H325c-6.5 0-10.3 7.4-6.5 12.7l124.6 172.8a31.8 31.8 0 0 0 51.7 0l210.6-292c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                    />
                  </svg>
                </i>
              </button>
            </li>
          </ul>
        </li>
        <li
          class="ant-list-item"
          data-testid="list-item"
        >
          <div
            class="ant-list-item-meta"
          >
            <div
              class="ant-list-item-meta-content"
            >
              <h4
                class="ant-list-item-meta-title"
              >
                <p
                  data-testid="item-p"
                  style="margin: 0px;"
                >
                  New task #2
                </p>
              </h4>
            </div>
          </div>
          <ul
            class="ant-list-item-action"
          >
            <li>
              <button
                class="ant-btn ant-btn-icon-only"
                data-testid="item-edit"
                style="color: rgb(255, 255, 255); background: rgb(124, 179, 5);"
                type="button"
              >
                <i
                  aria-label="icon: edit"
                  class="anticon anticon-edit"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="edit"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 0 0 0-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 0 0 9.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                    />
                  </svg>
                </i>
              </button>
              <em
                class="ant-list-item-action-split"
              />
            </li>
            <li>
              <button
                class="ant-btn ant-btn-icon-only"
                style="color: rgb(255, 255, 255); background: rgb(255, 120, 117);"
                type="button"
              >
                <i
                  aria-label="icon: delete"
                  class="anticon anticon-delete"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="delete"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                    />
                  </svg>
                </i>
              </button>
              <em
                class="ant-list-item-action-split"
              />
            </li>
            <li>
              <button
                class="ant-btn ant-btn-link"
                style="font-size: 20px;"
                type="button"
              >
                <i
                  aria-label="icon: check-circle"
                  class="anticon anticon-check-circle"
                  style="color: rgb(217, 217, 217);"
                  tabindex="-1"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="check-circle"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                      fill="#52c41a"
                    />
                    <path
                      d="M512 140c-205.4 0-372 166.6-372 372s166.6 372 372 372 372-166.6 372-372-166.6-372-372-372zm193.4 225.7l-210.6 292a31.8 31.8 0 0 1-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.3 0 19.9 5 25.9 13.3l71.2 98.8 157.2-218c6-8.4 15.7-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.4 12.7z"
                      fill="#f6ffed"
                    />
                    <path
                      d="M699 353h-46.9c-10.2 0-19.9 4.9-25.9 13.3L469 584.3l-71.2-98.8c-6-8.3-15.6-13.3-25.9-13.3H325c-6.5 0-10.3 7.4-6.5 12.7l124.6 172.8a31.8 31.8 0 0 0 51.7 0l210.6-292c3.9-5.3.1-12.7-6.4-12.7z"
                      fill="#52c41a"
                    />
                  </svg>
                </i>
              </button>
            </li>
          </ul>
        </li>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
